templates/pwm.asg.element.json

```diff
@@ -0,0 +1,1083 @@
+			"Description": "Name of the ALB PWM Target Group",
+			"Type": "String",
+			"MinLength": "1"
+		},
+		"PWMSNSTopicARN": {
+			"Description": "ARN of the SNS Topic to which to send auto scale action notifications",
+			"Type": "String"
+		},
+		"SecurityGroupIdPWMInstance": {
+			"Description": "ID of the security group for PWM instances",
+			"Type": "AWS::EC2::SecurityGroup::Id"
+		}
+	},
+	"Resources": {
+		"AllowAdminIAMGroupSSHKeyAccess": {
+			"Properties": {
+				"PolicyName": "AllowAdminIAMGroupSSHKeyAccess",
+				"PolicyDocument": {
+					"Statement": [{
+							"Action": [
+								"iam:GetGroup"
+							],
+							"Effect": "Allow",
+							"Resource": [{
+								"Fn::Join": [
+									"",
+									[
+										"arn:aws:iam::",
+										{
+											"Ref": "AWS::AccountId"
+										},
+										":group/",
+										{
+											"Ref": "AdminIAMGroup"
+										}
+									]
+								]
+							}]
+						},
+						{
+							"Action": [
+								"iam:GetSSHPublicKey",
+								"iam:ListSSHPublicKeys"
+							],
+							"Effect": "Allow",
+							"Resource": [{
+								"Fn::Join": [
+									"",
+									[
+										"arn:aws:iam::",
+										{
+											"Ref": "AWS::AccountId"
+										},
+										":user/*"
+									]
+								]
+							}]
+						}
+					],
+					"Version": "2012-10-17"
+				},
+				"Roles": [{
+					"Ref": "PWMInstanceRole"
+				}]
+			},
+			"Type": "AWS::IAM::Policy"
+		},
+		"AllowDiscoIAMGroupSSHKeyAccess": {
+			"Condition": "ConfigureDiscovery",
+			"Properties": {
+				"PolicyName": "AllowDiscoIAMGroupSSHKeyAccess",
+				"PolicyDocument": {
+					"Statement": [{
+							"Action": [
+								"iam:GetGroup"
+							],
+							"Effect": "Allow",
+							"Resource": [{
+								"Fn::Join": [
+									"",
+									[
+										"arn:aws:iam::",
+										{
+											"Ref": "AWS::AccountId"
+										},
+										":group/",
+										{
+											"Ref": "DiscoIAMGroup"
+										}
+									]
+								]
+							}]
+						},
+						{
+							"Action": [
+								"iam:GetSSHPublicKey",
+								"iam:ListSSHPublicKeys"
+							],
+							"Effect": "Allow",
+							"Resource": [{
+								"Fn::Join": [
+									"",
+									[
+										"arn:aws:iam::",
+										{
+											"Ref": "AWS::AccountId"
+										},
+										":user/*"
+									]
+								]
+							}]
+						}
+					],
+					"Version": "2012-10-17"
+				},
+				"Roles": [{
+					"Ref": "PWMInstanceRole"
+				}]
+			},
+			"Type": "AWS::IAM::Policy"
+		},
+		"CloudWatchAgent": {
+			"Properties": {
+				"Description": "CloudWatch Agent Server Policy for PWM instance role",
+				"PolicyDocument": {
+					"Statement": [{
+						"Action": [
+							"cloudwatch:PutMetricData",
+							"ec2:DescribeTags",
+							"logs:CreateLogGroup",
+							"logs:CreateLogStream",
+							"logs:DescribeLogStreams",
+							"logs:PutLogEvents",
+							"ssm:GetParameter"
+						],
+						"Effect": "Allow",
+						"Resource": [
+							"*"
```

Comment:
This is probably _far_ too broad a policy. At least these need to be resource-restricted:

* `cloudwatch:PutMetricData`
* `logs:PutLogEvents`
* `ssm:GetParameter`